hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SstFilteringService.java

```diff
@@ -231,4 +241,10 @@ public AtomicLong getSnapshotFilteredCount() {
   public BootstrapStateHandler.Lock getBootstrapStateLock() {
     return lock;
   }
+
+  @Override
+  public void shutdown() {
```

Comment:
Is this needed? As I mentioned above, the keymanager is already stopping the service.

Reply:
We need the sstFiltering service to be shutdown & all rocksdb instances to be closed before creating the hardlinks b/w the checkpoint directory & the actual follower meta directory. So yeah because of that we need to shutdown. Since we are going to get all the sst files from the leader anyways & going to restart the service anyhow we should start everything cleanslate. There is no point in running the loop till the end and shutting it down before will be more optmial

Reply:
> We need the sstFiltering service to be shutdown & all rocksdb
>   instances to be closed before creating the hardlinks b/w the
>   checkpoint directory & the actual follower meta directory.

This doesn't sound correct to me.  I'm not sure what "the actual follower meta directory" is, but the hard links are created where the tarball is untarred, (in the candidate directory.)   The SFS has no impact on it.

Then tarball data  gets linked over to the actual active fs in installCheckpoint() here: https://github.com/apache/ozone/blob/ffd84e747dc43d08a23269a8f411d52fef41c432/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L3696


That method also stops the keymanager here: https://github.com/apache/ozone/blob/ffd84e747dc43d08a23269a8f411d52fef41c432/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java#L3640

That is why these extra start/shutdown methods aren't needed.

Reply:
done resolving this